.pre-commit-config.yaml

```diff
@@ -0,0 +1,62 @@
+ci:
```

Comment:
Manual setup of pre-commit hooks (some still disabled to keep diff small for now)

Reply:
I am a bit torn on pre commit hooks. The reason we had a script + CI checks for linting is that sometimes, I have had the situation that black's autoformat completely destroyed my file. Typically, I do a commit after doing changes and then another commit (or multiple) for linting, such that there is a clean separation between linting and my stuff. If I understand it correctly, now we would enforce ruff being run before we can commit.  Or is it possible that we get a commit without the autoformatting + then a follow up commit with autolinting applied?

Reply:
`git commit -n` skips the pre-commit hooks. Is that what you are searching for? I also use it rather often locally when making a temporary snapshot of my code. If you don't like pre-commit hooks you can just not install them them :) We just have to define what the CI does and take that as source of truth.

Reply:
Ahh, I see. If we can easily skip the pre-commit hooks, then it sounds great!